### ISSUE TITLE
[Bug]: Jekyll 4.3.3 can't compile my scss

### ISSUE THREAD

**Issue by greatgraphicdesign** [jekyll/jekyll#9603]
### Operating System

macOS Ventura 13.6.6

### Ruby Version

ruby 3.3.1 (2024-04-23 revision c56cd86388) [x86_64-darwin22]

### Jekyll Version

jekyll 4.3.3

### GitHub Pages Version

Latest

### Expected Behavior

I expected my site to build successfully when running `bundle exec jekyll serve`.

### Current Behavior

Jekyll 4.22 was working fine in ruby 3.1.1, but I updated to ruby 3.3.1 and noticed that I didn't have the jekyll gem installed. I ran `bundle install` and `gem update` and then `bundle install` again. Here's my `Gemfile.lock`...

```
GEM
  remote: https://rubygems.org/
  specs:
    addressable (2.8.6)
      public_suffix (>= 2.0.2, < 6.0)
    bigdecimal (3.1.8)
    colorator (1.1.0)
    concurrent-ruby (1.3.1)
    em-websocket (0.5.3)
      eventmachine (>= 0.12.9)
      http_parser.rb (~> 0)
    eventmachine (1.2.7)
    ffi (1.16.3)
    forwardable-extended (2.6.0)
    google-protobuf (4.27.0-x86_64-darwin)
      bigdecimal
      rake (>= 13)
    http_parser.rb (0.8.0)
    i18n (1.14.5)
      concurrent-ruby (~> 1.0)
    jekyll (4.3.3)
      addressable (~> 2.4)
      colorator (~> 1.0)
      em-websocket (~> 0.5)
      i18n (~> 1.0)
      jekyll-sass-converter (>= 2.0, < 4.0)
      jekyll-watch (~> 2.0)
      kramdown (~> 2.3, >= 2.3.1)
      kramdown-parser-gfm (~> 1.0)
      liquid (~> 4.0)
      mercenary (>= 0.3.6, < 0.5)
      pathutil (~> 0.9)
      rouge (>= 3.0, < 5.0)
      safe_yaml (~> 1.0)
      terminal-table (>= 1.8, < 4.0)
      webrick (~> 1.7)
    jekyll-sass-converter (3.0.0)
      sass-embedded (~> 1.54)
    jekyll-watch (2.2.1)
      listen (~> 3.0)
    kramdown (2.4.0)
      rexml
    kramdown-parser-gfm (1.1.0)
      kramdown (~> 2.0)
    liquid (4.0.4)
    listen (3.9.0)
      rb-fsevent (~> 0.10, >= 0.10.3)
      rb-inotify (~> 0.9, >= 0.9.10)
    mercenary (0.4.0)
    pathutil (0.16.2)
      forwardable-extended (~> 2.6)
    public_suffix (5.0.5)
    rake (13.2.1)
    rb-fsevent (0.11.2)
    rb-inotify (0.11.1)
      ffi (~> 1.0)
    rexml (3.2.8)
      strscan (>= 3.0.9)
    rouge (4.2.1)
    safe_yaml (1.0.5)
    sass-embedded (1.77.3-x86_64-darwin)
      google-protobuf (>= 3.25, < 5.0)
    strscan (3.1.0)
    terminal-table (3.0.2)
      unicode-display_width (>= 1.1.1, < 3)
    unicode-display_width (2.5.0)
    webrick (1.8.1)

PLATFORMS
  x86_64-darwin-20
  x86_64-darwin-22

DEPENDENCIES
  jekyll
  webrick

BUNDLED WITH
   2.3.17
```

Jekyll breaks with the scss file in `/assets/css/main.scss`...

```
  Conversion error: Jekyll::Converters::Scss encountered an error while converting 'assets/css/main.scss':
                    expected "{".
                    ------------------------------------------------
      Jekyll 4.3.3   Please append `--trace` to the `serve` command 
                     for any additional information or backtrace. 
                    ------------------------------------------------
/Users/me/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/jekyll-sass-converter-3.0.0/lib/jekyll/converters/scss.rb:175:in `rescue in convert': expected "{". (Jekyll::Converters::Scss::SyntaxError)

        raise SyntaxError, e.message
```

Here's what the scss file contains (which worked fine previously)...
```
---
---

@import "fonts";
@import "main";
@import "colorbox";
@import "header";
@import "footer";
```

There's a red squiggly line under the first @import command. Hovering over it shows:
```
{ expectedscss(css-lcurlyexpected)
```

**Comment by greatgraphicdesign** [comment]
If I revert to ruby 3.1.1, replace the `Gemfile.lock` with the older version, and run `bundle install` then all is well. It would be nice to be able to run the latest versions though.

The `Gemfile.lock` file that works with ruby 3.1.1...
```
GEM
  remote: https://rubygems.org/
  specs:
    addressable (2.8.0)
      public_suffix (>= 2.0.2, < 5.0)
    colorator (1.1.0)
    concurrent-ruby (1.1.10)
    em-websocket (0.5.3)
      eventmachine (>= 0.12.9)
      http_parser.rb (~> 0)
    eventmachine (1.2.7)
    ffi (1.15.5)
    forwardable-extended (2.6.0)
    http_parser.rb (0.8.0)
    i18n (1.10.0)
      concurrent-ruby (~> 1.0)
    jekyll (4.2.2)
      addressable (~> 2.4)
      colorator (~> 1.0)
      em-websocket (~> 0.5)
      i18n (~> 1.0)
      jekyll-sass-converter (~> 2.0)
      jekyll-watch (~> 2.0)
      kramdown (~> 2.3)
      kramdown-parser-gfm (~> 1.0)
      liquid (~> 4.0)
      mercenary (~> 0.4.0)
      pathutil (~> 0.9)
      rouge (~> 3.0)
      safe_yaml (~> 1.0)
      terminal-table (~> 2.0)
    jekyll-sass-converter (2.2.0)
      sassc (> 2.0.1, < 3.0)
    jekyll-watch (2.2.1)
      listen (~> 3.0)
    kramdown (2.4.0)
      rexml
    kramdown-parser-gfm (1.1.0)
      kramdown (~> 2.0)
    liquid (4.0.3)
    listen (3.7.1)
      rb-fsevent (~> 0.10, >= 0.10.3)
      rb-inotify (~> 0.9, >= 0.9.10)
    mercenary (0.4.0)
    pathutil (0.16.2)
      forwardable-extended (~> 2.6)
    public_suffix (4.0.7)
    rb-fsevent (0.11.1)
    rb-inotify (0.10.1)
      ffi (~> 1.0)
    rexml (3.2.5)
    rouge (3.29.0)
    safe_yaml (1.0.5)
    sassc (2.4.0)
      ffi (~> 1.9)
    terminal-table (2.0.0)
      unicode-display_width (~> 1.1, >= 1.1.1)
    unicode-display_width (1.8.0)
    webrick (1.7.0)

PLATFORMS
  x86_64-darwin-20
  x86_64-darwin-22

DEPENDENCIES
  jekyll
  webrick

BUNDLED WITH
   2.3.17

```

**Comment by ntkme** [comment]
You have `@import "main";` inside `main.scss`, and this is no longer allowed. It worked in previously due side effect of a long existing bug, this bug has been fixed in jekyll-sass-converter 3.0 and this is now treated as circular import.

See https://github.com/jekyll/jekyll-sass-converter?tab=readme-ov-file#dropped-support-of-importing-files-with-the-same-filename-as-their-parent-file

**Comment by greatgraphicdesign** [comment]
Beautiful. Thank you for the solution! It works.
